File: Spring_Final_Project_04_01_26/Spring_Final_Project/src/main/java/com/My/Spring_Final_Project/Service/room/RoomTypeService.java
package com.My.Spring_Final_Project.Service.room;

import com.My.Spring_Final_Project.DTO.room.RoomTypeDTO;
import com.My.Spring_Final_Project.Entity.room.RoomType;

import java.util.List;

public interface RoomTypeService {
    RoomType create(RoomTypeDTO dto);
    List<RoomType> getAll();
}

File: Angular_Final_project/src/app/room/services/room-type.service.ts
// room-type.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { RoomType } from '../models/room-type.model';

@Injectable({
  providedIn: 'root'
})
export class RoomTypeService {
  private baseUrl = `${environment.apiUrl}/room-types`;

  constructor(private http: HttpClient) {}

  getAll(): Observable<RoomType[]> {
    return this.http.get<RoomType[]>(this.baseUrl);
  }

  create(roomType: RoomType): Observable<RoomType> {
    return this.http.post<RoomType>(this.baseUrl, roomType);
  }
}
